// vue-tg/dist/6.0.js
import { u as e, a as s, b as t, c as o, d as u, e as a, f as n, g as c, h as r, i, j as $, k as p, l as S, m as g, n as m, o as B, p as d, q as M, r as l, s as b, t as k } from "./useSecureStorage-BH7nD19q.mjs";
const y = () => e("6.0"), A = () => s("6.0"), C = () => t("6.0"), D = () => o("6.0"), H = () => u("6.0"), h = () => a("6.0"), j = () => n("6.0"), f = () => c("6.0"), w = () => r("6.0"), E = () => i("6.0"), F = () => $("6.0"), G = () => p("6.0"), L = () => S("6.0"), O = () => g("6.0"), P = () => m("6.0"), Q = () => B("6.0"), T = () => d("6.0"), V = () => M("6.0"), q = () => l("6.0"), x = () => b("6.0"), z = () => k("6.0");
export {
  y as useAccelerometer,
  A as useBackButton,
  C as useBiometricManager,
  D as useClipboard,
  H as useCloudStorage,
  h as useDeviceOrientation,
  j as useDeviceStorage,
  f as useEmojiStatus,
  w as useGyroscope,
  E as useHapticFeedback,
  F as useHomeScreen,
  G as useLocationManager,
  L as useMainButton,
  O as useMiniApp,
  P as usePopup,
  Q as useQrScanner,
  T as useSecondaryButton,
  V as useSecureStorage,
  q as useSettingsButton,
  x as useTheme,
  z as useViewport
};
