// vue-tg/dist/6.1.js
import { u as e, a as s, b as t, c as o, d as u, e as a, f as n, g as c, h as r, i, j as $, k as p, l as S, m as g, n as m, o as B, p as d, q as M, r as l, s as b, t as k } from "./useSecureStorage-BH7nD19q.mjs";
const y = () => e("6.1"), A = () => s("6.1"), C = () => t("6.1"), D = () => o("6.1"), H = () => u("6.1"), h = () => a("6.1"), j = () => n("6.1"), f = () => c("6.1"), w = () => r("6.1"), E = () => i("6.1"), F = () => $("6.1"), G = () => p("6.1"), L = () => S("6.1"), O = () => g("6.1"), P = () => m("6.1"), Q = () => B("6.1"), T = () => d("6.1"), V = () => M("6.1"), q = () => l("6.1"), x = () => b("6.1"), z = () => k("6.1");
export {
  y as useAccelerometer,
  A as useBackButton,
  C as useBiometricManager,
  D as useClipboard,
  H as useCloudStorage,
  h as useDeviceOrientation,
  j as useDeviceStorage,
  f as useEmojiStatus,
  w as useGyroscope,
  E as useHapticFeedback,
  F as useHomeScreen,
  G as useLocationManager,
  L as useMainButton,
  O as useMiniApp,
  P as usePopup,
  Q as useQrScanner,
  T as useSecondaryButton,
  V as useSecureStorage,
  q as useSettingsButton,
  x as useTheme,
  z as useViewport
};
