// vue-tg/dist/6.2.js
import { u as e, a as s, b as t, c as o, d as u, e as a, f as n, g as c, h as r, i, j as $, k as p, l as S, m as g, n as m, o as B, p as d, q as M, r as l, s as b, t as k } from "./useSecureStorage-BH7nD19q.mjs";
const y = () => e("6.2"), A = () => s("6.2"), C = () => t("6.2"), D = () => o("6.2"), H = () => u("6.2"), h = () => a("6.2"), j = () => n("6.2"), f = () => c("6.2"), w = () => r("6.2"), E = () => i("6.2"), F = () => $("6.2"), G = () => p("6.2"), L = () => S("6.2"), O = () => g("6.2"), P = () => m("6.2"), Q = () => B("6.2"), T = () => d("6.2"), V = () => M("6.2"), q = () => l("6.2"), x = () => b("6.2"), z = () => k("6.2");
export {
  y as useAccelerometer,
  A as useBackButton,
  C as useBiometricManager,
  D as useClipboard,
  H as useCloudStorage,
  h as useDeviceOrientation,
  j as useDeviceStorage,
  f as useEmojiStatus,
  w as useGyroscope,
  E as useHapticFeedback,
  F as useHomeScreen,
  G as useLocationManager,
  L as useMainButton,
  O as useMiniApp,
  P as usePopup,
  Q as useQrScanner,
  T as useSecondaryButton,
  V as useSecureStorage,
  q as useSettingsButton,
  x as useTheme,
  z as useViewport
};
