// vue-tg/dist/6.4.js
import { u as e, a as s, b as t, c as o, d as u, e as a, f as n, g as c, h as r, i, j as $, k as p, l as S, m as g, n as m, o as B, p as d, q as M, r as l, s as b, t as k } from "./useSecureStorage-BH7nD19q.mjs";
const y = () => e("6.4"), A = () => s("6.4"), C = () => t("6.4"), D = () => o("6.4"), H = () => u("6.4"), h = () => a("6.4"), j = () => n("6.4"), f = () => c("6.4"), w = () => r("6.4"), E = () => i("6.4"), F = () => $("6.4"), G = () => p("6.4"), L = () => S("6.4"), O = () => g("6.4"), P = () => m("6.4"), Q = () => B("6.4"), T = () => d("6.4"), V = () => M("6.4"), q = () => l("6.4"), x = () => b("6.4"), z = () => k("6.4");
export {
  y as useAccelerometer,
  A as useBackButton,
  C as useBiometricManager,
  D as useClipboard,
  H as useCloudStorage,
  h as useDeviceOrientation,
  j as useDeviceStorage,
  f as useEmojiStatus,
  w as useGyroscope,
  E as useHapticFeedback,
  F as useHomeScreen,
  G as useLocationManager,
  L as useMainButton,
  O as useMiniApp,
  P as usePopup,
  Q as useQrScanner,
  T as useSecondaryButton,
  V as useSecureStorage,
  q as useSettingsButton,
  x as useTheme,
  z as useViewport
};
