// vue-tg/dist/6.7.js
import { u as e, a as s, b as t, c as o, d as u, e as a, f as n, g as c, h as r, i, j as $, k as p, l as S, m as g, n as m, o as B, p as d, q as M, r as l, s as b, t as k } from "./useSecureStorage-BH7nD19q.mjs";
const y = () => e("6.7"), A = () => s("6.7"), C = () => t("6.7"), D = () => o("6.7"), H = () => u("6.7"), h = () => a("6.7"), j = () => n("6.7"), f = () => c("6.7"), w = () => r("6.7"), E = () => i("6.7"), F = () => $("6.7"), G = () => p("6.7"), L = () => S("6.7"), O = () => g("6.7"), P = () => m("6.7"), Q = () => B("6.7"), T = () => d("6.7"), V = () => M("6.7"), q = () => l("6.7"), x = () => b("6.7"), z = () => k("6.7");
export {
  y as useAccelerometer,
  A as useBackButton,
  C as useBiometricManager,
  D as useClipboard,
  H as useCloudStorage,
  h as useDeviceOrientation,
  j as useDeviceStorage,
  f as useEmojiStatus,
  w as useGyroscope,
  E as useHapticFeedback,
  F as useHomeScreen,
  G as useLocationManager,
  L as useMainButton,
  O as useMiniApp,
  P as usePopup,
  Q as useQrScanner,
  T as useSecondaryButton,
  V as useSecureStorage,
  q as useSettingsButton,
  x as useTheme,
  z as useViewport
};
