// vue-tg/dist/6.9.js
import { u as e, a as s, b as t, c as o, d as u, e as a, f as n, g as c, h as r, i, j as $, k as p, l as S, m as g, n as m, o as B, p as d, q as M, r as l, s as b, t as k } from "./useSecureStorage-BH7nD19q.mjs";
const y = () => e("6.9"), A = () => s("6.9"), C = () => t("6.9"), D = () => o("6.9"), H = () => u("6.9"), h = () => a("6.9"), j = () => n("6.9"), f = () => c("6.9"), w = () => r("6.9"), E = () => i("6.9"), F = () => $("6.9"), G = () => p("6.9"), L = () => S("6.9"), O = () => g("6.9"), P = () => m("6.9"), Q = () => B("6.9"), T = () => d("6.9"), V = () => M("6.9"), q = () => l("6.9"), x = () => b("6.9"), z = () => k("6.9");
export {
  y as useAccelerometer,
  A as useBackButton,
  C as useBiometricManager,
  D as useClipboard,
  H as useCloudStorage,
  h as useDeviceOrientation,
  j as useDeviceStorage,
  f as useEmojiStatus,
  w as useGyroscope,
  E as useHapticFeedback,
  F as useHomeScreen,
  G as useLocationManager,
  L as useMainButton,
  O as useMiniApp,
  P as usePopup,
  Q as useQrScanner,
  T as useSecondaryButton,
  V as useSecureStorage,
  q as useSettingsButton,
  x as useTheme,
  z as useViewport
};
