// vue-tg/dist/7.0.js
import { u as e, a as s, b as t, c as o, d as u, e as a, f as n, g as c, h as r, i, j as $, k as p, l as S, m as g, n as m, o as B, p as d, q as M, r as l, s as b, t as k } from "./useSecureStorage-BH7nD19q.mjs";
const y = () => e("7.0"), A = () => s("7.0"), C = () => t("7.0"), D = () => o("7.0"), H = () => u("7.0"), h = () => a("7.0"), j = () => n("7.0"), f = () => c("7.0"), w = () => r("7.0"), E = () => i("7.0"), F = () => $("7.0"), G = () => p("7.0"), L = () => S("7.0"), O = () => g("7.0"), P = () => m("7.0"), Q = () => B("7.0"), T = () => d("7.0"), V = () => M("7.0"), q = () => l("7.0"), x = () => b("7.0"), z = () => k("7.0");
export {
  y as useAccelerometer,
  A as useBackButton,
  C as useBiometricManager,
  D as useClipboard,
  H as useCloudStorage,
  h as useDeviceOrientation,
  j as useDeviceStorage,
  f as useEmojiStatus,
  w as useGyroscope,
  E as useHapticFeedback,
  F as useHomeScreen,
  G as useLocationManager,
  L as useMainButton,
  O as useMiniApp,
  P as usePopup,
  Q as useQrScanner,
  T as useSecondaryButton,
  V as useSecureStorage,
  q as useSettingsButton,
  x as useTheme,
  z as useViewport
};
