// vue-tg/dist/7.10.js
import { u as e, a as s, b as t, c as o, d as u, e as a, f as n, g as c, h as r, i, j as $, k as p, l as S, m as g, n as m, o as B, p as d, q as M, r as l, s as b, t as k } from "./useSecureStorage-BH7nD19q.mjs";
const y = () => e("7.10"), A = () => s("7.10"), C = () => t("7.10"), D = () => o("7.10"), H = () => u("7.10"), h = () => a("7.10"), j = () => n("7.10"), f = () => c("7.10"), w = () => r("7.10"), E = () => i("7.10"), F = () => $("7.10"), G = () => p("7.10"), L = () => S("7.10"), O = () => g("7.10"), P = () => m("7.10"), Q = () => B("7.10"), T = () => d("7.10"), V = () => M("7.10"), q = () => l("7.10"), x = () => b("7.10"), z = () => k("7.10");
export {
  y as useAccelerometer,
  A as useBackButton,
  C as useBiometricManager,
  D as useClipboard,
  H as useCloudStorage,
  h as useDeviceOrientation,
  j as useDeviceStorage,
  f as useEmojiStatus,
  w as useGyroscope,
  E as useHapticFeedback,
  F as useHomeScreen,
  G as useLocationManager,
  L as useMainButton,
  O as useMiniApp,
  P as usePopup,
  Q as useQrScanner,
  T as useSecondaryButton,
  V as useSecureStorage,
  q as useSettingsButton,
  x as useTheme,
  z as useViewport
};
