// vue-tg/dist/7.2.js
import { u as e, a as s, b as t, c as o, d as u, e as a, f as n, g as c, h as r, i, j as $, k as p, l as S, m as g, n as m, o as B, p as d, q as M, r as l, s as b, t as k } from "./useSecureStorage-BH7nD19q.mjs";
const y = () => e("7.2"), A = () => s("7.2"), C = () => t("7.2"), D = () => o("7.2"), H = () => u("7.2"), h = () => a("7.2"), j = () => n("7.2"), f = () => c("7.2"), w = () => r("7.2"), E = () => i("7.2"), F = () => $("7.2"), G = () => p("7.2"), L = () => S("7.2"), O = () => g("7.2"), P = () => m("7.2"), Q = () => B("7.2"), T = () => d("7.2"), V = () => M("7.2"), q = () => l("7.2"), x = () => b("7.2"), z = () => k("7.2");
export {
  y as useAccelerometer,
  A as useBackButton,
  C as useBiometricManager,
  D as useClipboard,
  H as useCloudStorage,
  h as useDeviceOrientation,
  j as useDeviceStorage,
  f as useEmojiStatus,
  w as useGyroscope,
  E as useHapticFeedback,
  F as useHomeScreen,
  G as useLocationManager,
  L as useMainButton,
  O as useMiniApp,
  P as usePopup,
  Q as useQrScanner,
  T as useSecondaryButton,
  V as useSecureStorage,
  q as useSettingsButton,
  x as useTheme,
  z as useViewport
};
