// vue-tg/dist/7.6.js
import { u as e, a as s, b as t, c as o, d as u, e as a, f as n, g as c, h as r, i, j as $, k as p, l as S, m as g, n as m, o as B, p as d, q as M, r as l, s as b, t as k } from "./useSecureStorage-BH7nD19q.mjs";
const y = () => e("7.6"), A = () => s("7.6"), C = () => t("7.6"), D = () => o("7.6"), H = () => u("7.6"), h = () => a("7.6"), j = () => n("7.6"), f = () => c("7.6"), w = () => r("7.6"), E = () => i("7.6"), F = () => $("7.6"), G = () => p("7.6"), L = () => S("7.6"), O = () => g("7.6"), P = () => m("7.6"), Q = () => B("7.6"), T = () => d("7.6"), V = () => M("7.6"), q = () => l("7.6"), x = () => b("7.6"), z = () => k("7.6");
export {
  y as useAccelerometer,
  A as useBackButton,
  C as useBiometricManager,
  D as useClipboard,
  H as useCloudStorage,
  h as useDeviceOrientation,
  j as useDeviceStorage,
  f as useEmojiStatus,
  w as useGyroscope,
  E as useHapticFeedback,
  F as useHomeScreen,
  G as useLocationManager,
  L as useMainButton,
  O as useMiniApp,
  P as usePopup,
  Q as useQrScanner,
  T as useSecondaryButton,
  V as useSecureStorage,
  q as useSettingsButton,
  x as useTheme,
  z as useViewport
};
