// vue-tg/dist/7.7.js
import { u as e, a as s, b as t, c as o, d as u, e as a, f as n, g as c, h as r, i, j as $, k as p, l as S, m as g, n as m, o as B, p as d, q as M, r as l, s as b, t as k } from "./useSecureStorage-BH7nD19q.mjs";
const y = () => e("7.7"), A = () => s("7.7"), C = () => t("7.7"), D = () => o("7.7"), H = () => u("7.7"), h = () => a("7.7"), j = () => n("7.7"), f = () => c("7.7"), w = () => r("7.7"), E = () => i("7.7"), F = () => $("7.7"), G = () => p("7.7"), L = () => S("7.7"), O = () => g("7.7"), P = () => m("7.7"), Q = () => B("7.7"), T = () => d("7.7"), V = () => M("7.7"), q = () => l("7.7"), x = () => b("7.7"), z = () => k("7.7");
export {
  y as useAccelerometer,
  A as useBackButton,
  C as useBiometricManager,
  D as useClipboard,
  H as useCloudStorage,
  h as useDeviceOrientation,
  j as useDeviceStorage,
  f as useEmojiStatus,
  w as useGyroscope,
  E as useHapticFeedback,
  F as useHomeScreen,
  G as useLocationManager,
  L as useMainButton,
  O as useMiniApp,
  P as usePopup,
  Q as useQrScanner,
  T as useSecondaryButton,
  V as useSecureStorage,
  q as useSettingsButton,
  x as useTheme,
  z as useViewport
};
