// vue-tg/dist/7.8.js
import { u as e, a as s, b as t, c as o, d as u, e as a, f as n, g as c, h as r, i, j as $, k as p, l as S, m as g, n as m, o as B, p as d, q as M, r as l, s as b, t as k } from "./useSecureStorage-BH7nD19q.mjs";
const y = () => e("7.8"), A = () => s("7.8"), C = () => t("7.8"), D = () => o("7.8"), H = () => u("7.8"), h = () => a("7.8"), j = () => n("7.8"), f = () => c("7.8"), w = () => r("7.8"), E = () => i("7.8"), F = () => $("7.8"), G = () => p("7.8"), L = () => S("7.8"), O = () => g("7.8"), P = () => m("7.8"), Q = () => B("7.8"), T = () => d("7.8"), V = () => M("7.8"), q = () => l("7.8"), x = () => b("7.8"), z = () => k("7.8");
export {
  y as useAccelerometer,
  A as useBackButton,
  C as useBiometricManager,
  D as useClipboard,
  H as useCloudStorage,
  h as useDeviceOrientation,
  j as useDeviceStorage,
  f as useEmojiStatus,
  w as useGyroscope,
  E as useHapticFeedback,
  F as useHomeScreen,
  G as useLocationManager,
  L as useMainButton,
  O as useMiniApp,
  P as usePopup,
  Q as useQrScanner,
  T as useSecondaryButton,
  V as useSecureStorage,
  q as useSettingsButton,
  x as useTheme,
  z as useViewport
};
