// vue-tg/dist/8.0.js
import { u as e, a as s, b as t, c as o, d as u, e as a, f as n, g as c, h as r, i, j as $, k as p, l as S, m as g, n as m, o as B, p as d, q as M, r as l, s as b, t as k } from "./useSecureStorage-BH7nD19q.mjs";
const y = () => e("8.0"), A = () => s("8.0"), C = () => t("8.0"), D = () => o("8.0"), H = () => u("8.0"), h = () => a("8.0"), j = () => n("8.0"), f = () => c("8.0"), w = () => r("8.0"), E = () => i("8.0"), F = () => $("8.0"), G = () => p("8.0"), L = () => S("8.0"), O = () => g("8.0"), P = () => m("8.0"), Q = () => B("8.0"), T = () => d("8.0"), V = () => M("8.0"), q = () => l("8.0"), x = () => b("8.0"), z = () => k("8.0");
export {
  y as useAccelerometer,
  A as useBackButton,
  C as useBiometricManager,
  D as useClipboard,
  H as useCloudStorage,
  h as useDeviceOrientation,
  j as useDeviceStorage,
  f as useEmojiStatus,
  w as useGyroscope,
  E as useHapticFeedback,
  F as useHomeScreen,
  G as useLocationManager,
  L as useMainButton,
  O as useMiniApp,
  P as usePopup,
  Q as useQrScanner,
  T as useSecondaryButton,
  V as useSecureStorage,
  q as useSettingsButton,
  x as useTheme,
  z as useViewport
};
