// vue-tg/dist/9.0.js
import { u as e, a as s, b as t, c as o, d as u, e as a, f as n, g as c, h as r, i, j as $, k as p, l as S, m as g, n as m, o as B, p as d, q as M, r as l, s as b, t as k } from "./useSecureStorage-BH7nD19q.mjs";
const y = () => e("9.0"), A = () => s("9.0"), C = () => t("9.0"), D = () => o("9.0"), H = () => u("9.0"), h = () => a("9.0"), j = () => n("9.0"), f = () => c("9.0"), w = () => r("9.0"), E = () => i("9.0"), F = () => $("9.0"), G = () => p("9.0"), L = () => S("9.0"), O = () => g("9.0"), P = () => m("9.0"), Q = () => B("9.0"), T = () => d("9.0"), V = () => M("9.0"), q = () => l("9.0"), x = () => b("9.0"), z = () => k("9.0");
export {
  y as useAccelerometer,
  A as useBackButton,
  C as useBiometricManager,
  D as useClipboard,
  H as useCloudStorage,
  h as useDeviceOrientation,
  j as useDeviceStorage,
  f as useEmojiStatus,
  w as useGyroscope,
  E as useHapticFeedback,
  F as useHomeScreen,
  G as useLocationManager,
  L as useMainButton,
  O as useMiniApp,
  P as usePopup,
  Q as useQrScanner,
  T as useSecondaryButton,
  V as useSecureStorage,
  q as useSettingsButton,
  x as useTheme,
  z as useViewport
};
