// vue-tg/dist/9.1.js
import { u as e, a as s, b as t, c as o, d as u, e as a, f as n, g as c, h as r, i, j as $, k as p, l as S, m as g, n as m, o as B, p as d, q as M, r as l, s as b, t as k } from "./useSecureStorage-BH7nD19q.mjs";
const y = () => e("9.1"), A = () => s("9.1"), C = () => t("9.1"), D = () => o("9.1"), H = () => u("9.1"), h = () => a("9.1"), j = () => n("9.1"), f = () => c("9.1"), w = () => r("9.1"), E = () => i("9.1"), F = () => $("9.1"), G = () => p("9.1"), L = () => S("9.1"), O = () => g("9.1"), P = () => m("9.1"), Q = () => B("9.1"), T = () => d("9.1"), V = () => M("9.1"), q = () => l("9.1"), x = () => b("9.1"), z = () => k("9.1");
export {
  y as useAccelerometer,
  A as useBackButton,
  C as useBiometricManager,
  D as useClipboard,
  H as useCloudStorage,
  h as useDeviceOrientation,
  j as useDeviceStorage,
  f as useEmojiStatus,
  w as useGyroscope,
  E as useHapticFeedback,
  F as useHomeScreen,
  G as useLocationManager,
  L as useMainButton,
  O as useMiniApp,
  P as usePopup,
  Q as useQrScanner,
  T as useSecondaryButton,
  V as useSecureStorage,
  q as useSettingsButton,
  x as useTheme,
  z as useViewport
};
